PocketKit/Sources/PocketKit/Refresh/HomeRefreshCoordinator.swift

```diff
@@ -36,18 +36,30 @@ class HomeRefreshCoordinator: HomeRefreshCoordinatorProtocol {
         }
 
         if shouldRefresh(isForced: isForced), !isRefreshing {
-            Task {
+            Task { [weak self] in
+                guard let self = self else {
```

Comment:
you could make this more compact, omitting `= self`  if you like: `guard let self else { ...`